### PR TITLE
Two AMLA changes

### PR DESCRIPTION
**Pull request by CelticMinstrel**
The first change (using `[modifications][advancement]` instead of `[modifications][advance]`) makes it possible to use the same macro (eg `{AMLA_DEFAULT}`) both for adding an advancement possibility to a unit type and for adding that an advancement to a specific placed unit. The implementation is designed to be backwards compatible.

The second change allows making AMLA possibilities mutually exclusive. There is currently no way to indicate to the player that this is the case though; it would be nice to have some indication so that the player knows that certain choices will be eliminated by the choice they've made. However, I think that's a separate issue (in fact, it already affects 1.12 if you use strict_amla=no).

### CODE REVIEW

File: src/scripting/game_lua_kernel.cpp [hunk 1]
```diff
@@ -3276,7 +3276,9 @@ static int intf_add_modification(lua_State *L)
 	unit_ptr u = luaW_checkunit(L, 1);
 	char const *m = luaL_checkstring(L, 2);
 	std::string sm = m;
-	if (sm != "advance" && sm != "object" && sm != "trait")
+	if (sm == "advance") // Maintain backwards compatibility
+		sm = "advancement";
```
Comment (shikadilord) [inline src/scripting/game_lua_kernel.cpp]: Need braces around if body block.

File: src/unit.cpp [hunk 2]
```diff
@@ -1574,24 +1575,40 @@ std::vector<config> unit::get_modification_advances() const
...
+		{
+			int max_num = std::count(temp_exclude.begin(), temp_exclude.end(), s);
+			int mod_num = modification_count("advancement", s);
+			if (mod_num >= max_num) {
+				exclusion_found = true;
+				break;
+			}
+		}
+		if (exclusion_found)
+			continue;
```
Comment (shikadilord) [inline src/unit.cpp]: Need braces around if body block.

**Comment by ln-zookeeper** [comment]
Can one not currently add [advancement]s to a single unit to give it extra advancement options?

How does the game (or the WML author for that matter) distinguish between a single unit [advancement] which adds an advancement option and another which specifies an advancement the unit has already gained, if both use the same tag?

**Comment by CelticMinstrel** [comment]
The distinction between an advancement opportunity and an advancement already gained is placement - those already gained are a child of the [modifications] tag, while advancement opportunities would end up being siblings of [modifications].

**Comment by CelticMinstrel** [comment]
I addressed the style issues @shikadilord raised by amending the relevant commits.

**Comment by shikadilord** [comment]
Just missing warnings for instances of the old tag name, as I mentioned on IRC. It looks good to me otherwise.

**Pull request merged**